File: subastas/src/main/java/org/example/subastas/domain/Subasta.java
package org.example.subastas.domain;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class Subasta {

    private Long id;

    private String title;

    private String description;

    private Categoria categoria;

    private BigDecimal precioInicial;

    private LocalDateTime fechaInicio;

    private LocalDateTime fechaFin;

    private Long userId;
}

File: subastasFront/src/app/subasta.ts
import { Categoria } from "./categoria";

export interface Subasta {

    id: number;
    title: string;
    description: string;
    categoria: string;
    precioInicial: number;
    fechaInicio: string;
    ubicacion: string;
    fechaFin: string;
    martilleroId: number | null;
    imagenes: string[];
    estado: string;
    userId: number;
    userGanadorId: number | null;
    incrementoFijo: number;
}
